deployments/local-minio/deploy.sh

```diff
@@ -1,64 +1,118 @@
 #!/usr/bin/env bash
 
 # Name to give the docker network and stack. Update network instances in docker-compose.yml file if modified.
-networkname="cdr_local-minio"
+name="cdr_local-minio"
 
+helptext="\
+Usage: deploy.sh [command]
+Utility script for setting up a docker network and deploying a local instance of CDR into a docker swarm.
+'deploy' is the default option if no command is given.
+
+Commands:
+  d, deploy\t\tcreate the docker network and stack, stopping if one already exists
+  u, update\t\tattempts an in-place update of the deployed docker stack
+  c, clean\t\tremoves the deployed docker stack and network (destructive)
+  r, redeploy\t\tclean + deploy: completely removes the stack and network and deploys new instances (destructive)
+"
+
+# Removes the docker stack and network.
+# Docker networks usually take a few seconds to go down, so it waits to prevent redeployment issues
+function cleanup () {
+    printf "Removing $name docker stack and network...\n"
+    docker stack rm $name
+    docker network rm $name
+    waiting=$(docker network ls | grep "$name")
+    printf "\nWaiting for docker network '$name' to go down."
+    while [ ! -z "$waiting" ]; do
+        sleep 2
+        printf "."
+        waiting=$(docker network ls | grep " $name ")
+    done
+    printf "\nDone!\n"
+}
+
+# Waits for everything to start up
 function wait_for_containers () {
-    waiting=$(docker service ls | grep ""$networkname"_.*0/1 ")
-    printf "\nWaiting for docker services to start."
+    waiting=$(docker service ls | grep ""$name"_.*0/1 ")
+    printf "\nWaiting for docker services to start (safe to exit)."
     while [ ! -z "$waiting" ]; do
         sleep 2
         printf "."
-        waiting=$(docker service ls | grep ""$networkname"_.*0/1 ")
+        waiting=$(docker service ls | grep ""$name"_.*0/1 ")
     done
     printf "\nDone!\n"
 }
 
 # Deploy docker-compose.yml file via docker stacks
 function deploy_images () {
-    printf "Deploying Docker Images...\n"
-    docker stack deploy -c docker-compose.yml $networkname
+    printf "\nDeploying Docker Images...\n"
+    exists=$(docker stack ls | grep "$name")
+    if [ ! -z "$exists" ]; then
+        printf "[ERROR] '$name' stack already exists. Please run script with 'clean', 'update', 'redeploy', or 'help' for more info.\n"
```

Comment:
:exclamation: If the stack exists, the update option will always print "... stack already exists" and exit.
Try executing "./deploy.sh" followed by "./deploy.sh update"

Reply:
Nice catch! I'll add an update method that has a positive check for a pre-existing stack.